ERCS/erc-7546.md

```diff
@@ -0,0 +1,178 @@
+---
+eip: 7546
+title: Upgradeable Clone for Scalable Contracts
+description: An upgradeable, cloneable, horizontally extensible proxy pattern.
+author: Shogo Ochiai (@shogochiai) <shogo.ochiai@pm.me>, Kai Hiroi (@KaiHiroi) <kai.hiroi@pm.me>
+discussions-to: https://ethereum-magicians.org/t/eip-7546-upgradeable-clone/16256
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-10-25
+requires: 165, 1967, 7201
+---
+
+## Abstract
+It has been a significant challenge for developers attempting to create cloneable and upgradeable contracts on the Ethereum Virtual Machine (EVM). While UUPS and the Diamond Standard offer partial solutions, a comprehensive answer has remained elusive. Our proposal addresses this gap through the introduction of two main features.
```

Comment:
Please expand uncommon abbreviations the first time they are used.

---

The first approach is to link to the proposal defining UUPS. This approach has the downside of preventing your proposal from advancing statuses until ERC-1822 advances.

```suggestion
It has been a significant challenge for developers attempting to create cloneable and upgradeable contracts on the Ethereum Virtual Machine (EVM). While [ERC-1822](./eip-1822.md)'s UUPS and [ERC-2535](./eip-2535.md) Diamonds offer partial solutions, a comprehensive answer has remained elusive. Our proposal addresses this gap through the introduction of two main features.
```

---

Alternatively, you can just omit UUPS:

```suggestion
It has been a significant challenge for developers attempting to create cloneable and upgradeable contracts on the Ethereum Virtual Machine (EVM). While [ERC-2535](./eip-2535.md) Diamonds and other existing proxy standards offer partial solutions, a comprehensive answer has remained elusive. Our proposal addresses this gap through the introduction of two main features.
```

Reply:
Any time you link to another proposal, either in the `requires` header or in the markdown body. See [here](https://ethereum.github.io/eipw/markdown-link-status/) for a brief explanation.